vlog4j-graal/src/main/java/org/semanticweb/vlog4j/graal/GraalToVLog4JModelConverter.java

```diff
@@ -0,0 +1,219 @@
+package org.semanticweb.vlog4j.graal;
+
+/*-
+ * #%L
+ * VLog4J Graal Import Components
+ * %%
+ * Copyright (C) 2018 VLog4j Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeAtom;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConjunction;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConstant;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makePredicate;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeRule;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeVariable;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.semanticweb.vlog4j.core.model.api.Atom;
+import org.semanticweb.vlog4j.core.model.api.Conjunction;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Term;
+
+import fr.lirmm.graphik.graal.api.core.AtomSet;
+import fr.lirmm.graphik.graal.api.core.ConjunctiveQuery;
+import fr.lirmm.graphik.util.stream.CloseableIterator;
+import fr.lirmm.graphik.util.stream.IteratorException;
+
+/**
+ * Utility class to convert Graal data structures ("model classes") into VLog4J
+ * data structures. Labels ({@link ConjunctiveQuery#getLabel()},
+ * {@link fr.lirmm.graphik.graal.api.core.Rule#getLabel() Rule.getLabel()}, or
+ * {@link fr.lirmm.graphik.graal.api.core.Term#getLabel() Term.getLabel()}) are
+ * not converted since VLog4J does not support them.
+ *
+ * @author Adrian Bielefeldt
+ *
+ */
+public final class GraalToVLog4JModelConverter {
+
+	private GraalToVLog4JModelConverter() {
+	};
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atom} into a
+	 * {@link Atom VLog4J Atom}.
+	 *
+	 * @param atom A {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atom}
+	 * @return A {@link Atom VLog4J Atom}
+	 */
+	public static Atom convertAtom(final fr.lirmm.graphik.graal.api.core.Atom atom) {
```

Comment:
vlog4j does not support predicates with arity <1, blank predicate names, atoms with no terms or null terms, rules with empty heads or bodies, null atoms within rules.  An exception (NullPointerException, InvalidArgumentException) is thrown when creating such object in vlog4j core. However, especially when converting a graal ConjunctiveQuery to a vlog4j rule, these cases (at least empty answer variables list, empty query body) should be validated during conversion, and a GraalConvertException should be thrown.